python/cudf/cudf/_lib/csv.pyx

```diff
@@ -232,37 +240,67 @@ cdef csv_reader_options make_csv_reader_options(
 
     if dtype is not None:
         if isinstance(dtype, abc.Mapping):
-            c_dtypes.reserve(len(dtype))
             for k, v in dtype.items():
-                c_dtypes.push_back(
-                    str(
-                        str(k)+":"+
-                        _get_cudf_compatible_str_from_dtype(v)
-                    ).encode()
-                )
+                col_type = v
+                if v in {"hex", "hex64"}:
+                    c_hex_col_names.push_back(str(k).encode())
+                    col_type = 'int64'
+                elif v == "hex32":
+                    c_hex_col_names.push_back(str(k).encode())
+                    col_type = 'int32'
+
+                c_dtypes_map[str(k).encode()] = \
+                    _get_cudf_data_type_from_dtype(
+                        cudf.dtype(col_type))
+            csv_reader_options_c.set_dtypes(c_dtypes_map)
+            csv_reader_options_c.set_parse_hex(c_hex_col_names)
         elif (
             cudf.utils.dtypes.is_scalar(dtype) or
             isinstance(dtype, (
                 np.dtype, pd.core.dtypes.dtypes.ExtensionDtype, type
             ))
         ):
-            c_dtypes.reserve(1)
-            c_dtypes.push_back(
-                _get_cudf_compatible_str_from_dtype(dtype).encode()
-            )
-        elif isinstance(dtype, abc.Iterable):
-            c_dtypes.reserve(len(dtype))
-            for col_dtype in dtype:
-                c_dtypes.push_back(
-                    _get_cudf_compatible_str_from_dtype(col_dtype).encode()
+            c_dtypes_list.reserve(1)
+            if dtype in {"hex", "hex64"}:
+                c_hex_col_indexes.push_back(0)
+                c_dtypes_list.push_back(
+                    _get_cudf_data_type_from_dtype('int64')
+                )
+            elif dtype == "hex32":
+                c_hex_col_indexes.push_back(0)
+                c_dtypes_list.push_back(
+                    _get_cudf_data_type_from_dtype('int32')
                 )
+            else:
+                c_dtypes_list.push_back(
+                    _get_cudf_data_type_from_dtype(dtype)
+                )
+            csv_reader_options_c.set_dtypes(c_dtypes_list)
+            csv_reader_options_c.set_parse_hex(c_hex_col_indexes)
+        elif isinstance(dtype, abc.Iterable):
+            c_dtypes_list.reserve(len(dtype))
+            for index, col_dtype in enumerate(dtype):
+                if col_dtype in {"hex", "hex64"}:
+                    c_hex_col_indexes.push_back(index)
+                    c_dtypes_list.push_back(
+                        _get_cudf_data_type_from_dtype('int64')
+                    )
+                elif col_dtype == "hex32":
+                    c_hex_col_indexes.push_back(index)
+                    c_dtypes_list.push_back(
+                        _get_cudf_data_type_from_dtype('int32')
+                    )
```

Comment:
variations of this block appear three times, wondering if the core logic can be factored out (namely, the hex->int mapping)

Reply:
Moved the type mapping to a separate dict that helped to remove lot of the code.

Reply:
Moved the type mapping to a separate dict that helped to remove lot of the code.

Reply:
That looks great!